# HG changeset patch
# Parent  7a6d836b62779aa61988981c6ca646495574a505
Bug 1314968 - Explicitly specify the AccessPoint interface name. r?kanru

The DBus specification allows passing an empty string as the interface to the
org.freedesktop.DBus.Properties.GetAll call to get all properties, throwing away the namespace
(interface) information.

However, GDBus does not allow this. When NetworkManager moved to using GDBus, Firefox lost the
ability to retrieve access points from NetworkManager.

Since we're only interested in properties from the org.freedesktop.NetworkManager.AccessPoint
interface, name it explicitly. This works with both the old and the new NetworkManager.

diff --git a/netwerk/wifi/nsWifiScannerDBus.cpp b/netwerk/wifi/nsWifiScannerDBus.cpp
--- a/netwerk/wifi/nsWifiScannerDBus.cpp
+++ b/netwerk/wifi/nsWifiScannerDBus.cpp
@@ -151,7 +151,7 @@ nsWifiScannerDBus::SendGetAll(const char
   DBusMessageIter argsIter;
   dbus_message_iter_init_append(msg, &argsIter);
 
-  const char* param = "";
+  const char* param = "org.freedesktop.NetworkManager.AccessPoint";
   if (!dbus_message_iter_append_basic(&argsIter, DBUS_TYPE_STRING, &param)) {
     return NS_ERROR_FAILURE;
   }
